src/XrdPosix/XrdPosixMap.cc

```diff
@@ -80,6 +80,41 @@ mode_t XrdPosixMap::Flags2Mode(dev_t *rdv, uint32_t flags)
 
    return newflags;
 }
+
+/******************************************************************************/
+/*                             S t a t F i l l                                */
+/******************************************************************************/
+
+int XrdPosixMap::Entry2Buf(const XrdCl::DirectoryList::ListEntry &dirEnt, struct stat &buf)
+{
+   auto statInfo = dirEnt.GetStatInfo();
+   if (!statInfo) return EIO;
+
+   memset(&buf, '\0', sizeof(buf));
+   buf.st_mode   = XrdPosixMap::Flags2Mode(nullptr, statInfo->GetFlags());
+   // Since the UID/GID isn't known by the client, when these are translated by
+   // XrdXrootdProtocol::StatGen back to xrootd protocol flags, they will get zero'd
```

Comment:
xroot is the protocol and xrootd is the implementation of the protocol. Please correct.